docker/entrypoint.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/sh
+
+# usage: file_env VAR [DEFAULT]
+#    ie: file_env 'XYZ_DB_PASSWORD' 'example'
+# (will allow for "$XYZ_DB_PASSWORD_FILE" to fill in the value of
+#  "$XYZ_DB_PASSWORD" from a file, especially for Docker's secrets feature)
+file_env() {
+    local var="$1"
+    local fileVar="${var}_FILE"
+    local def="${2:-}"
+    if [ "${!var:-}" ] && [ "${!fileVar:-}" ]; then
+        echo >&2 "error: both $var and $fileVar are set (but are exclusive)"
+        exit 1
+    fi
+    local val="$def"
+    if [ "${!var:-}" ]; then
+        val="${!var}"
+    elif [ "${!fileVar:-}" ]; then
+        val="$(< "${!fileVar}")"
+    fi
+    export "$var"="$val"
+    unset "$fileVar"
+}
+
+file_env 'IOTA_AUTH_USER'
+file_env 'IOTA_AUTH_PASSWORD'
+file_env 'IOTA_AUTH_ENABLED'
+
+if [[ $IOTA_AUTH_ENABLED ==  "false" ]]; then
```

Comment:
hmmm... this still doesn't quite make sense, you're saying if I **deliberately set** the `ENV` to `false` then the component should nag you, but `false` is the default anyway so why would anyone set it?

It should be a empty check not an equals check - If I deliberately override the `ENV` to switch off security I don't need to be nagged, if I **forget** to switch it on I do.

Reply:
Isn't this the way to check if auth is enabled or not as suggested in comment https://github.com/telefonicaid/iotagent-json/pull/380#discussion_r254249337 ?

Reply:
No - read it carefully - emphasis added:

> * Then for users in secure mode only check if  IOTA_AUTH_USER + IOTA_AUTH_PASSWORD **are set (not blank)** and echo the other warning.

Reply:
Fixed with https://github.com/telefonicaid/iotagent-json/pull/380/commits/c33b866240d67dcd3071bf277b278e1e617c5c9f